packages/client/hmi-client/src/temp/tera-simulation-workflow.vue

```diff
@@ -1,22 +1,86 @@
 <template>
-	<infinite-canvas debug-mode :last-transform="lastTransform" @save-transform="saveTransform">
+	<infinite-canvas
+		debug-mode
+		@contextmenu="toggleContextMenu"
+		:last-transform="canvasTransform"
+		@save-transform="saveTransform"
+	>
 		<template #foreground></template>
 		<template #data>
-			<div style="font-size: 24px; padding: 10px; background: #9ef">This is a DataLayer DIV</div>
+			<ContextMenu ref="contextMenu" :model="contextMenuItems" />
+
+			<ul v-for="node in nodes">
+				<tera-workflow-node :node="node" :initialTransform="canvasTransform"></tera-workflow-node>
+			</ul>
 		</template>
 	</infinite-canvas>
 </template>
 
 <script setup lang="ts">
+import { ref } from 'vue';
 import InfiniteCanvas from '@/components/widgets/tera-infinite-canvas.vue';
+import { Operation, WorkflowNode, WorkflowStatus } from '@/types/workflow';
+import TeraWorkflowNode from './tera-workflow-node.vue';
+import ContextMenu from 'primevue/contextmenu';
+
+const nodes = ref<WorkflowNode[]>([]);
+const contextMenu = ref();
+const newNodePosition = ref<{ x: number; y: number }>({ x: 0, y: 0 });
+const canvasTransform = ref<d3.ZoomTransform>();
```

Comment:
Make this of type `{ k: number; x: number; y: number }` (same as `lastTransform` prop in the infinite canvas). Also in `saveTransform()` see about extracting the x, y, z from the `d3.ZoomTransform` type and assigning it to the `canvasTransform`.  Or better yet in `handleZoomEnd` pass `{ e.transform.x, e.transform.y, e.transform.k }` to `save-transform`. That way we just have to deal with d3 types in the infinite canvas.

Reply:
Good point!